packages/design-system-react-native/src/components/Button/variants/ButtonPrimary/ButtonPrimary.tsx

```diff
@@ -110,14 +111,15 @@ const ButtonPrimaryBase = ({
   );
 };
 
-const ButtonPrimaryLightOnly = withThemeProvider(
-  ButtonPrimaryBase,
-  Theme.Light,
-);
-
 export const ButtonPrimary = ({ isInverse, ...props }: ButtonPrimaryProps) => {
+  // If inverse, use the current theme context
   if (isInverse) {
-    return <ButtonPrimaryBase isInverse={isInverse} {...props} />;
+    return <ButtonPrimaryBase isInverse {...props} />;
   }
-  return <ButtonPrimaryLightOnly isInverse={isInverse} {...props} />;
+  // Otherwise, force light theme
+  return (
+    <ThemeProvider theme={Theme.Light}>
+      <ButtonPrimaryBase isInverse={false} {...props} />
+    </ThemeProvider>
```

Comment:
Wraps the `ButtonPrimaryBase` component in its own ThemeProvider set to the light theme, but only when isInverse is not set. In this case, the button will always use the light theme, no matter what theme is set at the app level. React supports nested providers, and this is how we should intentionally override context values for a subtree.